### PR TITLE
Discovery app revisited - Side by Side Navigation

### PR DESCRIPTION
**Pull request by Khaledgarbaya**
Restyling Navigation, So that content types is always present when you are navigating between entries.

side-by-side-navigation

### CODE REVIEW

File: src/components/List.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React, {PropTypes} from 'react'
+import CSSModules from 'react-css-modules'
+import styles from './List.css'
+
+const listContainerStyle = {
+  height: (window.innerHeight - 200) + 'px'
+}
```
Comment (jelz) [inline src/components/List.js:7]: Put an empty line here (and below: after component, after prop types)

File: src/components/List.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+class List extends React.Component {
+  componentWillMout () {
```
Comment (jelz) [inline src/components/List.js:10]: Is this needed?

File: src/components/TwoPanelList.js [hunk 3]
```diff
@@ -1,34 +1,26 @@
...
+import List from './List'
 
-function TwoPanelList ({items, ListView, NavView, TitleView, ContentView, ListActionView, location}) {
-  const list = items.map((item) => <ListView key={item.sys.id} item={item} location={location}/>)
-  const listContainerStyle = {
-    height: (window.innerHeight - 200) + 'px'
-  }
-
+function TwoPanelList ({items, ContentView, location}) {
+  const lists = []
```
Comment (jelz) [inline src/components/TwoPanelList.js]: You can `const lists = items.map((...`

File: src/components/TwoPanelList.js [hunk 4]
```diff
@@ -1,34 +1,26 @@
...
+  items.forEach((topItem, index) => {
+    const list = topItem.items.map((innerItem) => <topItem.ListView key={innerItem.sys.id} item={innerItem} location={location}/>)
```
Comment (jelz) [inline src/components/TwoPanelList.js]: Please make this and next line shorter (e.g. break line on arrow and props)

File: src/components/App.css [hunk 5]
```diff
@@ -53,3 +53,13 @@
   padding-top: 40px;
   composes: centered-container;
 }
+.logo{
```
Comment (jelz) [inline src/components/App.css]: Is this needed?

File: src/components/entries/EntriesContainer.js [hunk 6]
```diff
@@ -88,8 +90,8 @@ export default createClass({
       return <p>No entries are available.</p>
     } else {
```
Comment (jelz) [inline src/components/entries/EntriesContainer.js:97]: You're `return`ing in both `if` and `else if` clauses. You can get rid of `else` as the code here is rather long.

File: src/components/entries/EntriesContainer.js [hunk 7]
```diff
@@ -88,8 +90,8 @@ export default createClass({
...
       let contentElement, loadMoreElement
-      const backNavLink = <Link to={{pathname: '/entries/by-content-type', query: this.props.location.query}}>&lt; Content Type List</Link>
-      const listTitle = <h3>{this.state.entries.getAt(0).sys.contentType.name}</h3>
+      const contentTypeListTitle = <h3>Content Types</h3>
```
Comment (jelz) [inline src/components/entries/EntriesContainer.js]: No need for that, you use it only once. The same goes for entry list title.

File: src/components/entries/EntriesContainer.js [hunk 8]
```diff
@@ -99,14 +101,10 @@ export default createClass({
       if (this.state.entries.len() < this.state.total) {
         loadMoreElement = <FeaturelessButton label='Load more' action={this.loadEntries}/>
       }
-
       return <TwoPanelList
-        items={this.state.entries.value}
-        ListView={EntryListItem}
-        NavView={backNavLink}
-        TitleView={listTitle}
+        items={[{items: this.state.contentTypes, TitleView: contentTypeListTitle, ListView: ContentTypeListItem},
```
Comment (jelz) [inline src/components/entries/EntriesContainer.js]: You can extract these objects.

File: src/components/List.js (hunk 2 above)
Comment (Khaledgarbaya) [inline src/components/List.js:10]: this is to make the navigation expand all the way to the bottom, was there from the last version but I moved it to the List component, I guess I can use CSS to do that also

Comment (jelz) [inline src/components/List.js:10]: I'm asking if `componentWillMout` method is needed here.

**Pull request merged**